stm32/powerctrlboot.c: Updatde clock and PLL selection.

chrismas9 opened the pull request:
This addresses https://github.com/micropython/micropython/issues/5049
The baud rate was wrong because stm32lib SystemCoreClockUpdate sets
SystemCoreClock to 8MHz instead of 48MHz if HSI48 is routed directly to SYSCLK.
Work around is to use HSI48 -> PREDIV (/2) -> PLL (*2) -> SYSCLK.

chrismas9 committed:
stm32/powerctrlboot.c: Updated clock and PLL selection.
This addresses https://github.com/micropython/micropython/issues/5049
The baud rate was wrong because stm32lib SystemCoreClockUpdate sets
SystemCoreClock to 8MHz instead of 48MHz if HSI48 is routed directly to SYSCLK.
Work around is to use HSI48 -> PREDIV (/2) -> PLL (*2) -> SYSCLK.

dpgeorge commented:
Good find!  Maybe there's a reason the HAL doesn't support HSI48 as the direct SYSCLK...

dpgeorge commented on ports/stm32/powerctrlboot.c:
```diff
@@ -38,11 +38,20 @@ void SystemClock_Config(void) {
-    const uint32_t sysclk_src = 3;
+    RCC->CFGR = 0 << RCC_CFGR_PLLMUL_Pos | 3 << RCC_CFGR_PLLSRC_Pos; // PLL mult by 2, src = HSI48/PREDIV
+    RCC->CFGR2 = 1; // Input clock divided by 2
+    RCC->CR |= RCC_CR_PLLON; // Turn PLL on
```
It'd be good not to duplicate this PLL code, since it's also used in the HSE part below.

So the `#endif` below could be moved up, to just after the `RCC->CFGR2 = 0;` line.

chrismas9 committed:
Add STM32F0 support for HSI at 8MHz -> PLL x6 = 48 MHz for MCUs without HSI48.
Select clock source explicitly in mpconfigboard.h
Use HSE Bypass when HSE is chosen. NUCLEO clock source is STLINK, not crystal.

chrismas9 commented:
> Maybe there's a reason the HAL doesn't support HSI48 as the direct SYSCLK...

`HAL_RCC_GetSysClockFreq` returns the correct value with HSI48 direct. STMCubeMX allows it and I HSI48 appears to meet all clock specs for peripherals. Latest HAL drivers from STMCubeMX project have the same bug. I think it is a bug or oversight in the HAL.

dpgeorge commented:
eb12fa38624a6bb4b71ef72f2aadc0f7ca8f27d1Thank you.  Merged in f16e4be3fa16a3532bb93ccf88e867d27fe2ff20 and
